File: invoiceJavaBackend/src/main/java/com/example/invoiceJavaBackend/dto/InvoiceDTO.java
package com.example.invoiceJavaBackend.dto;

import java.math.BigDecimal;
import java.util.Date;

import javax.validation.constraints.NotEmpty;

public class InvoiceDTO {
    
    @NotEmpty
    private String invoiceNumber;

    @NotEmpty
    private String type;
    
    @NotEmpty
    private String placeOfIssue;

    private Date dateOfIssue;

    private Date dateOfSale;

    @NotEmpty
    private String methodOfPayment;

    private String dateOfPayment;

    private String accountNumber;

    private BigDecimal totalValue;

    private String remarks;

    public InvoiceDTO() {
        
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(String accountNumber) {
        this.accountNumber = accountNumber;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getInvoiceNumber() {
        return invoiceNumber;
    }

    public String getRemarks() {
        return remarks;
    }

    public void setRemarks(String remarks) {
        this.remarks = remarks;
    }

    public BigDecimal getTotalValue() {
        return totalValue;
    }

    public void setTotalValue(BigDecimal totalValue) {
        this.totalValue = totalValue;
    }

    public String getDateOfPayment() {
        return dateOfPayment;
    }

    public void setDateOfPayment(String dateOfPayment) {
        this.dateOfPayment = dateOfPayment;
    }

    public String getMethodOfPayment() {
        return methodOfPayment;
    }

    public void setMethodOfPayment(String methodOfPayment) {
        this.methodOfPayment = methodOfPayment;
    }

    public Date getDateOfSale() {
        return dateOfSale;
    }

    public void setDateOfSale(Date dateOfSale) {
        this.dateOfSale = dateOfSale;
    }

    public Date getDateOfIssue() {
        return dateOfIssue;
    }

    public void setDateOfIssue(Date dateOfIssue) {
        this.dateOfIssue = dateOfIssue;
    }

    public String getPlaceOfIssue() {
        return placeOfIssue;
    }

    public void setPlaceOfIssue(String placeOfIssue) {
        this.placeOfIssue = placeOfIssue;
    }

    public void setInvoiceNumber(String invoiceNumber) {
        this.invoiceNumber = invoiceNumber;
    }

}

File: InvoiceSharpWPFApp/InvoiceSharpWPFApp/Dto/InvoiceDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InvoiceSharpWPFApp.Dto
{
    public class InvoiceDTO
    {
        public InvoiceDTO()
        {
            // setting default values
            type = "Outgoing";
            dateOfIssue = DateTime.Now;
            dateOfSale = DateTime.Now;
            methodOfPayment = "Cash";
            totalValue = 0.00M;
        }

        public string invoiceNumber { get; set; }

        public string type { get; set; }

        public string placeOfIssue { get; set; }

        public DateTime dateOfIssue { get; set; }

        public DateTime dateOfSale { get; set; }

        public string methodOfPayment { get; set; }

        public string dateOfPayment { get; set; }

        public string accountNumber { get; set; }

        public decimal totalValue { get; set; }

        public string remarks { get; set; }

    }
}
